File: flib/src/main/java/com/f1reking/flib/base/BaseLibFragment.java
package com.f1reking.flib.base;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

/**
 * @author F1ReKing
 * @date 2019/10/16 13:39
 * @Description
 */
public abstract class BaseLibFragment extends Fragment {

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(getLayoutId(), container, false);
        initView(view);
        return view;
    }

    abstract int getLayoutId();

    abstract void initView(View view);
}

File: flib/src/main/java/com/f1reking/flib/base/BaseLibFragment.kt
package com.f1reking.flib.base

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment

/**
 * @author: F1ReKing
 * @date: 2019/3/28 23:07
 * @desc:
 */
abstract class BaseLibFragment : Fragment() {

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val view = inflater.inflate(getLayoutId(), container, false)
        return view
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        initView()
    }

    abstract fun initView()

    abstract fun getLayoutId(): Int
}